File: Examples/Ruby/Annotations/get_page_annotations.rb
require 'aspose_pdf_cloud'

class Annotation

  include AsposePDFCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @pdf_api = PdfApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../data/" << file_name,"r") { |io| io.read } )
  end

  def get_page_annotations
    file_name = "Sample-Annotation.pdf"
    upload_file(file_name)

    page_number = 1
    response = @pdf_api.get_page_annotations(file_name, page_number)
  end

end

annotation = Annotation.new()
puts annotation.get_page_annotations

File: Examples/PHP/Annotations/GetPageAnnotations.php
<?php
require_once realpath(__DIR__ . '/..') . '/vendor/autoload.php';
require_once realpath(__DIR__ . '/..') . '/Utils.php';

use Aspose\PDF\PdfApi;
use Aspose\PDF\AsposeApp;

class Annotation {

	public $pdfApi;

	public function __construct() {
		AsposeApp::$appSID = Utils::appSID;
		AsposeApp::$apiKey = Utils::apiKey;
		$this->pdfApi = new PdfApi();
	}

	public function getPageAnnotations() {
		// Upload file to Aspose Cloud Storage
		$fileName = "Sample-Annotation.pdf";
		Utils::uploadFile($fileName);

		$result = $this->pdfApi->GetPageAnnotations($fileName, $pageNumber = 1, $storage = "", $folder = "");
		print_r ( $result );
	}
}

$annotation = new Annotation();
$annotation->getPageAnnotations();

?>
